tools/ci.sh

```diff
@@ -966,7 +969,7 @@ function ci_zephyr_run_tests {
     docker exec zephyr-ci west build -p auto -b qemu_cortex_m3 -- -DCONF_FILE=prj_minimal.conf
     # Issues with zephyr tests:
     # - inf_nan_arith fails pow(-1, nan) test
```

Comment:
As above, can we move this check into the test?

Reply:
I think there's actually a bug in zephyr's libm code, meaning `pow(-1, nan)` doesn't return the correct value.

I don't think our `MICROPY_PY_MATH_POW_FIX_NAN` option fixes that, we would need to add another such workaround.  Probably worth doing.